onnxruntime/test/contrib_ops/qlinear_where_test.cc

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#include "gtest/gtest.h"
+#include "test/providers/provider_test_utils.h"
+#include <array>
+
+namespace onnxruntime {
+namespace test {
+
+template <typename T>
+void RunQLinearWhere(
+    OpTester &test,
+    const std::vector<T> &x,
+    const std::vector<T> &y,
+    const std::vector<T> &z,
+    float x_scale,
+    T x_zero_point,
+    float y_scale,
+    T y_zero_point,
+    float z_scale,
+    T z_zero_point,
+    const std::vector<int64_t> &x_shape,
+    const std::vector<int64_t> &y_shape,
+    const std::vector<int64_t> &z_shape) {
+
+  test.AddInput<T>("X", x_shape, x);
+  test.AddInput<float>("x_scale", {}, {x_scale}, true);
+  test.AddInput<T>("x_zero_point", {}, {x_zero_point}, true);
+  test.AddInput<T>("Y", y_shape, y);
+  test.AddInput<float>("y_scale", {}, {y_scale}, true);
+  test.AddInput<T>("y_zero_point", {}, {y_zero_point}, true);
+  test.AddInput<float>("z_scale", {}, {z_scale});
+  test.AddInput<T>("z_zero_point", {}, {z_zero_point});
+  test.AddOutput<T>("Z", z_shape, z);
+  test.Run(OpTester::ExpectResult::kExpectSuccess);
+}
+template <typename T>
+void QLinearWhereScalarAll() {
+  OpTester test("QLinearWhere", 1, onnxruntime::kMSDomain);
+  test.AddInput<bool>("condition", {1}, {true}, true  );
+  RunQLinearWhere<T>(
+      test,
+      {1}, {2}, {1},// x ,y ,z
+      1.0f, 0, 1.0f, 0, 1.0f, 0, // x_scale, x_zp, y_scale, y_zp, z_scale, z_zp
```

Comment:
>1.0f, 0, 1.0f, 0, 1.0f, 0, // x_scale, x_zp, y_scale, y_zp, z_scale, z_zp [](http://example.com/codeflow?start=6&length=73)

Please use a more general scale and zp instead of 1.0 and 0.